src/qmapshack/gis/rte/router/CRouterRoutino.cpp

```diff
@@ -269,12 +296,29 @@ void CRouterRoutino::freeDatabaseList()
 {
     for(int i = 0; i < comboDatabase->count(); i++)
     {
-        Routino_Database * data = (Routino_Database*)comboDatabase->itemData(i, Qt::UserRole).toULongLong();
+        QVariantMap map = comboDatabase->itemData(i,Qt::UserRole).toMap();
+        Routino_Database * data = (Routino_Database*)(map["db"].toULongLong());
         Routino_UnloadDatabase(data);
     }
     comboDatabase->clear();
 }
 
+void CRouterRoutino::loadProfiles(QString profilesPath)
+{
+    // load xml profiles if it is new before setting the profile!
+    if( currentProfilesPath != profilesPath)
+    {
+        currentProfilesPath = profilesPath;
+        int res = 0;
+        res = Routino_ParseXMLProfiles(profilesPath.toStdString().data());
```

Comment:
please use `toUtf8()` as in the original call.

Reply:
Ok, but remove the commented line of code.

Reply:
Done